proposals/csharp-9.0/native-integers.md

```diff
@@ -43,34 +43,39 @@ There is an identity conversion between compound types that differ by native int
 The tables below cover the conversions between special types.
 (The IL for each conversion includes the variants for `unchecked` and `checked` contexts if different.)
 
+General notes:
+- Most docs explain `conv.u` as conversion to __native unsigned integer__ and `conv.i` as conversion to __native integer__. But a more appropriate description is that `conv.u` is zero-extending and `conv.i` is sign-extending (if the most significant bit is `1`, then the conversion fills with `1`s).
+- Some non-obvious choices are guided by pre-existing semantics. For example, `sbyte->nint` uses `conv.i` and `byte->nint` uses `conv.u`, the same way that `sbyte->long` uses `conv.i8` and `byte->long` uses `conv.u8`. Also, `checked(void*->nint)` uses `conv.ovf.i.un` the same way that `checked(void*->long)` uses `conv.ovf.i8.un`.
```

Comment:
I think you've got this inversed.

`sbyte->nint` and `sbyte->nuint` use `conv.i` while `byte->nint` and `byte->nuint` use `conv.u` because they are all **widening** and therefore the `i` effectively means `sign-extend` while the `u` effectively means `zero-extend`.

`nint->byte` and `nuint->byte` use `conv.u1` while `nint->sbyte` and `nuint->sbyte` use `conv.i1` as a `pre-existing semantic`. These are both **narrowing** and so there isn't a specific need to pick `signed` (`i1`) or `unsigned` (`u1`), its just a semantic of what's already done.

The overflow cases do have semantics that are important and that's `conv.ovf.*` where the input is treated as `signed` and the output is `*` and `conv.ovf.*.un` where the input is treated as `unsigned` and the output is `*`.

Reply:
So basically, `checked` contexts for both **widening** and **narrowing** are:
* `conv.ovf.*` for `signed to *`
* `conv.ovf.*.un` for `unsigned to *`

While, `unchecked` contexts for **widening** are:
* `conv.i*` for `signed to *` (where * is the target width)
* `conv.u*` for `unsigned to *` (where * is the target width)

And, `unchecked` contexts for **narrowing** are:
* `conv.i*` for `any to signed *` (where * is the target width)
* `conv.u*` for `any to unsigned *` (where * is the target width)

Reply:
> so there isn't a specific need to pick signed (i1) or unsigned (u1), its just a semantic of what's already done.

After re-reading the ECMA-335 bit, there is actually an important semantic here for **narrowing**.

For `byte`, `sbyte`, `short`, and `ushort` the "stack type" is `int32`. So `conv.i1` is effectively "downcast to a signed byte and then sign-extend up to int32" while `conv.u1` is effectively "downcast to an unsigned byte and then zero-extend up to int32"